ucc/tests/test_compile.py

```diff
@@ -88,47 +88,93 @@ def test_tket_compile():
     assert isinstance(result_circuit, TketCircuit)
 
 
+def test_custom_pass():
+    """Verify that a custom pass works with a non-qiskit input circuit"""
+
+    class HtoX(TransformationPass):
+        """Toy transformation that converts all H gates to X gates"""
+
+        def run(self, dag):
+            for node in dag.op_nodes():
+                if not isinstance(node.op, HGate):
+                    continue
+                dag.substitute_node(node, XGate())
+            return dag
+
+    # Example usage with a cirq circuit, stil showcasing the cross-frontend compatibility
+
+    qubit = NamedQubit("q_0")
+    cirq_circuit = CirqCircuit(H(qubit))
+
+    post_compiler_circuit = compile(cirq_circuit, custom_passes=[HtoX()])
+    assert_same_circuits(post_compiler_circuit, CirqCircuit(X(qubit)))
+
+
 def test_compile_with_target_device():
     circuit = QiskitCircuit(3)
     circuit.cx(0, 1)
     circuit.cx(0, 2)
 
     # Create a simple target that does not have direct CX between 0 and 2
-    t = Target(description="Fake device", num_qubits=3)
-    t.add_instruction(CXGate(), {(0, 1): None, (1, 2): None})
+    t = Mybackend().target
     result_circuit = compile(
         circuit, return_format="original", target_device=t
     )
 
-    # Check compilation respected the target device topology
     dag = circuit_to_dag(result_circuit)
+
+    # Check that the gateset is the same as the target device
+    gateset = set()
+    for node in dag.op_nodes():
+        gateset.add(node.op.name)
+
+    # Check that each gate in the compiled gateset is available in the target device
+    for gate in gateset:
+        assert gate in t.operation_names
+
+    # Check that the coupling map of the compiled circuit is the same as the target device
     analysis_pass = CheckMap(
         t.build_coupling_map(), property_set_field="check_map"
     )
     analysis_pass.run(dag)
     assert analysis_pass.property_set["check_map"]
 
 
-def test_custom_pass():
-    """Verify that a custom pass works with a non-qiskit input circuit"""
+def test_return_gateset():
```

Comment:
To be consistent with above test:
```suggestion
def test_compile_with_return_gateset():
```